src/fenicsxconcrete/sensor_definition/reaction_force_sensor.py

```diff
@@ -49,21 +49,37 @@ def measure(self, problem: MaterialProblem, t: float = 1.0) -> None:
         reaction_force_vector = []
 
         bc_generator_x = BoundaryConditions(problem.mesh, problem.V)
-        bc_generator_x.add_dirichlet_bc(df.fem.Constant(domain=problem.mesh, c=1.0), self.surface, 0, "geometrical", 0)
+        bc_generator_x.add_dirichlet_bc(
+            value=df.fem.Constant(domain=problem.mesh, c=1.0),
+            boundary=self.surface,
+            sub=0,
+            method="geometrical",
+            entity_dim=problem.p["dim"] - 1,
```

Comment:
Suggestion: would it make sense to distinguish between topological dim and geometrical dim of the problem?
For example, with Gmsh you can create a mesh of a Line that is embedded in 2D space and therefore has `tdim=1` and `gdim=2`.